modules/videoio/src/cap_ffmpeg_impl.hpp

```diff
@@ -1402,20 +1404,26 @@ bool CvCapture_FFMPEG::grabFrame()
     return valid;
 }
 
-bool CvCapture_FFMPEG::retrieveFrame(int, unsigned char** data, int* step, int* width, int* height, int* cn)
+bool CvCapture_FFMPEG::retrieveFrame(int flag, unsigned char** data, int* step, int* width, int* height, int* cn)
 {
     if (!video_st)
         return false;
 
-    if (rawMode)
+    if (rawMode || flag == extraDataIdx)
     {
-        AVPacket& p = bsfc ? packet_filtered : packet;
-        *data = p.data;
-        *step = p.size;
-        *width = p.size;
+        if (flag == 0) {
+            AVPacket& p = bsfc ? packet_filtered : packet;
+            *data = p.data;
+            *step = p.size;
+        }
+        else if (flag == extraDataIdx) {
+            *data = ic->streams[video_stream]->codec->extradata;
+            *step = ic->streams[video_stream]->codec->extradata_size;
+        }
+        *width = *step;
         *height = 1;
         *cn = 1;
-        return p.data != NULL;
+        return *data != NULL;
```

Comment:
> `return *data != NULL;`

Why we fail even if codec extradata is empty? (e.g., not needed with used video stream)

Reply:
> e.g., not needed with used video stream
Sorry alalek can you clarify?  If we request extra data but it is not available shouldn't the function return false?

Reply:
"return false;" means error here (C style coding rudiment - exception with proper message is more suitable here).
Empty data is empty cv::Mat buffer.